airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py

```diff
@@ -1175,3 +1179,170 @@ async def test_streams_define_primary_key(
+
+        with open(docs_path, "r") as docs_file:
+            prereq_content_lines = docs_file.readlines()[prereq_start_line:prereq_end_line]
+            prereq_content = "".join(prereq_content_lines).lower()
+            required_titles, has_credentials = docs_utils.required_titles_from_spec(actual_connector_spec.connectionSpecification)
+
+            for title in required_titles:
+                assert title in prereq_content, f"Required '{title}' field is not in {self.PREREQUISITES} section."
+
+            if has_credentials:
+                credentials_validation = [k in prereq_content for k in self.CREDENTIALS_KEYWORDS]
+                assert True in credentials_validation, f"Required 'credentials' field is not in {self.PREREQUISITES} section."
+
+    def test_docs_structure(self, connector_documentation: str, connector_metadata: dict):
+        node = docs_utils.documentation_node(connector_documentation)
+        heading_names = tuple([docs_utils.remove_step_from_heading(docs_utils.header_name(n)) for n in node if n.type == self.HEADING])
+        template_headings, non_required_heading = self._get_template_headings(connector_metadata["data"]["name"])
+
+        heading_names_len, template_headings_len = len(heading_names), len(template_headings)
+        heading_names_index, template_headings_index = 0, 0
+
+        while heading_names_index < heading_names_len and template_headings_index < template_headings_len:
+            heading_names_value = heading_names[heading_names_index]
+            template_headings_value = template_headings[template_headings_index]
+
+            if template_headings_value == self.CONNECTOR_SPECIFIC_HEADINGS:
+                if heading_names_value not in template_headings:
+                    heading_names_index += 1
+                    continue
+                else:
+                    template_headings_index += 1
+                    continue
+            if heading_names_value == template_headings_value:
+                heading_names_index += 1
+                template_headings_index += 1
+                continue
+            if template_headings_value in non_required_heading:
+                template_headings_index += 1
+                continue
+
+            pytest.fail(
+                f"Documentation structure doesn't follow standard template."
+                f" Heading '{heading_names_value}' is not on a right place, name of heading is incorrect or heading name is not expected.\n"
+                f"Diff:\nCurrent Heading: '{heading_names_value}'. Expected Heading: '{template_headings_value}'"
+            )
+
+        if template_headings_index != template_headings_len:
+            pytest.fail(
+                f"Documentation structure doesn't follow standard template. docs is not full.\nMissing headers: {template_headings[template_headings_index:]}"
+            )
+
+    def test_docs_descriptions(self, docs_path: str, connector_documentation: str, connector_metadata: dict):
```

Comment:
Can we test this some additional certified connectors like `source-greenhouse`, `source-airtable` and a few others. I'm a little concerned how many connectors will fail this validation since it's quite specific to the exact text

Reply:
Yes sure, I will test it with some more certified connectors and let you know the results. My previous tests, unfortunately , show differences in the actual and expected docs structure, but it was reasonable, errors like typos, incorrect order of heading etc were found using this test suite.

Reply:
[here](https://docs.google.com/document/d/1mGKXy0fqIZ8sGdyjyU0ps5AVkAapdsUaMuQdyGr6D5E/edit?usp=sharing) is my test results for some connectors

Reply:
I don't have access to this document. Please share with me